detox/src/devices/drivers/android/genycloud/GenyCloudDriver.js

```diff
@@ -69,12 +71,15 @@ class GenyCloudDriver extends AndroidDriver {
   }
 
   async cleanup(instance, bundleId) {
-    await this._deviceRegistry.disposeDevice(instance.uuid);
-    await super.cleanup(instance, bundleId);
+      try {
+        await super.cleanup(instance, bundleId);
+      } finally {
+        await this._instanceAllocation.deallocateDevice(instance.uuid);
+      }
   }
 
   async shutdown(instance) {
-    await this._instanceAllocation.deallocateDevice(instance);
+    await this._instanceLauncher.shutdown(instance);
```

Comment:
`GenyCloudInstanceLauncher.shutdown()` now executes the logic previously implemented in `GenyCloudInstanceAllocation.deallocateDevice`